### PR TITLE
refactor(@angular/cli): rename config file from angular-cli.json to .…

### PR DESCRIPTION
**Pull request by Brocco**
…angular-cli.json

### CODE REVIEW

File: docs/design/ngConfig.md [hunk 1]
```diff
@@ -16,14 +16,14 @@ Since the data is static, we only need to keep it in a static store somewhere.
 
 One solution would be to keep the data in the `package.json`. Unfortunately, the metadata contains too much data and the `package.json` file would become unmanageable.
 
-Instead of polluting the package file, a `angular-cli.json` file will be created that contains all the values. Access to that file will be allowed to the user if he knows the structure of the file (unknown keys will be kept but ignored), and it's easy to read and write.
+Instead of polluting the package file, a `.angular-cli.json` file will be created that contains all the values. Access to that file will be allowed to the user if he knows the structure of the file (unknown keys will be kept but ignored), and it's easy to read and write.
```
Comment (johnpapa) [inline docs/design/ngConfig.md]: avoid `he` and `she`. be gender indifferent. Instead use `who` here.

File: docs/design/ngConfig.md [hunk 2]
```diff
@@ -16,14 +16,14 @@ Since the data is static, we only need to keep it in a static store somewhere.
...
 
 
 ## Fallback
 
-There should be two `angular-cli.json` files; one for the project and a general one. The general one should contain information that can be useful when scaffolding new apps, or informations about the user.
+There should be two `.angular-cli.json` files; one for the project and a general one. The general one should contain information that can be useful when scaffolding new apps, or informations about the user.
```
Comment (johnpapa) [inline docs/design/ngConfig.md]: `informations` --> `information`

File: docs/documentation/build.md [hunk 3]
```diff
@@ -19,7 +19,7 @@ The build artifacts will be stored in the `dist/` directory.
 environment file to be used with that build (`--environment=dev` or `--environment=prod`).
 By default, the development build target and environment are used.
 
-The mapping used to determine which environment file is used can be found in `angular-cli.json`:
+The mapping used to determine which environment file is used can be found in `.angular-cli.json`:
```
Comment (johnpapa) [inline docs/documentation/build.md]: It might be nice to show the schema or a link to it somewhere.

File: docs/documentation/build.md [hunk 4]
```diff
@@ -46,7 +46,7 @@ ng build
 
 You can also add your own env files other than `dev` and `prod` by doing the following:
```
Comment (johnpapa) [inline docs/documentation/build.md]: maybe say environment files instead of env files.

File: packages/@angular/cli/models/config.ts [hunk 5]
```diff
@@ -5,7 +5,8 @@ import * as chalk from 'chalk';
 import * as fs from 'fs';
 import * as path from 'path';
 
-export const CLI_CONFIG_FILE_NAME = 'angular-cli.json';
+export const CLI_CONFIG_FILE_NAME = '.angular-cli.json';
+const CLI_CONFIG_FILE_NAME_ALT = 'angular-cli.json';
```
Comment (filipesilva) [inline packages/@angular/cli/models/config.ts]: Should we show a warning if users are using the old one?

Comment (Brocco) [inline packages/@angular/cli/models/config.ts]: I spoke with @hansl about this, and we think we should silently fallback to the old file name. This means fewer changes are needed, but newly generated projects will generate with the new file name.

**Comment by Brocco** [comment]
@johnpapa I agree with the suggestions you provided, but this PR is to change the file name, I will create a new PR with the language suggestions.

**Pull request merged**